shell/edit/fleet.cattle.io.gitrepo.vue

```diff
@@ -289,6 +291,10 @@ export default {
 
     updateCachedAuthVal(val, key) {
       this.tempCachedValues[key] = typeof val === 'string' ? { selected: val } : { ...val };
+
+      if (key === 'helmSecretName') {
+        this.displayhelmRepoURLRegex = val && val.selected !== AUTH_TYPE._NONE;
```

Comment:
Does it make sense to have watch on `displayhelmRepoURLRegex` when this is the only place we change it?
We also repeat the check logic in `updatehelmRepoURLRegex`, while we already are assigning it here.